pkg/localstore/mode_put.go

```diff
@@ -466,6 +466,11 @@ func (db *DB) addToCache(
 	batch *leveldb.Batch,
 	item shed.Item,
 ) (int64, error) {
+
+	err := db.pullIndex.DeleteInBatch(batch, item)
```

Comment:
So moving these doesnt make any difference. The index is only updated when the Batch is written. So order doesnt matter.

Reply:
Here we have early returns from the function due to the pin check, this should not affect the chunk being removed from the reserve.

Reply:
So on L481 we return if there is an existing pinIndex. In this case, we dont put the item to gcIndex and hence we should not delete the pullIndex.